core/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -458,4 +459,14 @@ public void setEagerGlobalOrdinals(boolean eagerGlobalOrdinals) {
         this.eagerGlobalOrdinals = eagerGlobalOrdinals;
     }
 
+    /** Return a {@link ValueFormat} that can be used to display and parse
+     *  values as returned by the fielddata API.
+     *  The default implementation returns a {@link ValueFormat.Raw}. */
+    public ValueFormat valueFormat(@Nullable String format, DateTimeZone timeZone) {
+        if (format != null) {
+            throw new IllegalArgumentException("Field [" + name() + "] of type [" + typeName() + "] does not support custom formats");
+        }
```

Comment:
Good question. Currently the code always passes a non-null value (UTC by default) but I guess you are right that it would make more sense to treat both arguments tho same way.